neo/io/baseio.py

```diff
@@ -96,7 +98,7 @@ class BaseIO:
 
     mode = 'file'  # or 'fake' or 'dir' or 'database'
 
-    def __init__(self, filename=None, **kargs):
+    def __init__(self, filename: str | Path = None, **kargs):
```

Comment:
The use of `|` in type hints was added in Python 3.10. Since we still support Python 3.8 and 3.9, this should be `Union[str, Path]`